app/src/main/java/com/mamadou/newsapp/networking/RemoteApi.kt

```diff
@@ -0,0 +1,36 @@
+package com.mamadou.newsapp.networking
+
+import com.mamadou.newsapp.models.Article
+import com.mamadou.newsapp.models.response.GetNewsResponse
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+const val BASE_URL ="https://newsapi.org"
+const val TOKEN_KEY = "926a9e3d32f545be911ca1e71f9aff72"
+class RemoteApi(private val apiService: RemoteApiService) {
+
+    fun getArticles(onArticlesReceived: (List<Article>, Throwable?) -> Unit) {
+        apiService.getArticles(TOKEN_KEY).enqueue(object : Callback<GetNewsResponse> {
+            override fun onResponse(
+                call: Call<GetNewsResponse>,
+                response: Response<GetNewsResponse>
+            ) {
+                val data = response.body()
+
+                if (data != null && data.articles.isNotEmpty()) {
+                    println("******* Call Request Received $response *********")
+                    println("********* See Articles Below $data **********")
+                    println("*********** Number Of Articles are ${response.body()!!.articles.size} ***********")
```

Comment:
very minor nit pick, `println` works fine but if you want more Logcat granulator (printing to INFO, DEBUG, WARNING or ERROR outputs), you could use the `Log` class.

Reply:
I use Log.i for the positive response and Log.e for the error messages.